pkgs/servers/monitoring/plugins/zfs.nix

```diff
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, python3
+, zfs
+, sudo
+}:
+
+let
+  n="check_zfs";
+in stdenv.mkDerivation {
+  pname = "check-zfs";
+  version = "2.0";
+
+  src = fetchFromGitHub {
+    owner  = "zlacelle";
+    repo   = "nagios_check_zfs_linux";
+    rev    = "33bea0671953a8891b46db6951f56c59a120a16c";
+    sha256 = "gPLCNt6hp4E94s9/PRgsnBN5XXQQ+s2MGcgRFeknXg4=";
+  };
+
+  buildInputs = [python3 zfs sudo];
+
+  installPhase = ''
+    mkdir -p $out/bin
+    sed -i 's#/usr/bin/python3#${python3}/bin/python#g' check_zfs.py
+    sed -i 's#/usr/bin/sudo#${sudo}/bin/sudo#g' check_zfs.py
+    sed -i 's#/sbin/zpool#${zfs}/bin/zpool#g' check_zfs.py
+    sed -i 's#/sbin/zfs#${zfs}/bin/zfs#g' check_zfs.py
+    cp ${n}.py $out/bin/${n}
+    chmod +x $out/bin/${n}
+  '';
```

Comment:
Please add pre/post hooks.

Please use `install(1)` instead of `mkdir`/`cp` to install files into the derivation output. You can pass `-m 755` to make a file executable.

Please use `substituteInPlace` instead of `sed`.